tls/s2n_resume.h

```diff
@@ -69,6 +69,8 @@ struct s2n_ticket_key_weight {
 struct s2n_ticket_fields {
     struct s2n_blob session_secret;
     uint32_t ticket_age_add;
+    uint64_t current_time;
```

Comment:
This isn't really the "current_time", given that you've stored it and have no idea when it will be accessed. You need to be really clear on what this value actually means. "key_retrieval_time" might be more accurate, if I'm understanding your logic right.

Is there a reason your code needs to use "key_retrieval_time" instead of just a true "now"?

You mention your [alternative branch](https://github.com/aws/s2n-tls/compare/main...boquan-fang:s2n-tls:serialize-refactor#). If the problem is the repetition of retrieving the key and the time, is there a way you can fix that problem without abandoning the solution altogether?